test/test_kfac.py

```diff
@@ -200,3 +206,71 @@ def test_kfac_inplace_activations(dev: device):
     ggn_no_inplace = ggn_block_diagonal(model, loss_func, params, data)
 
     report_nonclose(ggn, ggn_no_inplace)
+
+
+@mark.parametrize("fisher_type", ["type-2", "mc", "empirical"])
+@mark.parametrize("loss", [MSELoss, CrossEntropyLoss], ids=["mse", "ce"])
+@mark.parametrize("reduction", ["mean", "sum"])
+@mark.parametrize("dev", DEVICES, ids=DEVICES_IDS)
+def test_multi_dim_output(
+    fisher_type: str,
+    loss: Union[MSELoss, CrossEntropyLoss],
+    reduction: str,
+    dev: device,
+):
+    """Test the KFAC implementation for >2d outputs (using a 3d and 4d output).
+
+    Args:
+        fisher_type: The type of Fisher matrix to use.
+        loss: The loss function to use.
+        reduction: The reduction to use for the loss function.
+        dev: The device to run the test on.
+    """
+    manual_seed(0)
+    # set up loss function, data, and model
+    loss_func = loss(reduction=reduction).to(dev)
+    if isinstance(loss_func, MSELoss):
+        data = [
+            (rand(2, 4, 5), regression_targets((2, 4, 3))),
+            (rand(4, 7, 5, 5), regression_targets((4, 7, 5, 3))),
+        ]
```

Comment:
There is a problem with feeding such data.
`curvlinops` assumes that all batches in the data set can be stacked along the batch axis.
If this is not the case, `curvlinops` cannot figure out how to properly accumulate mini-batch losses into the loss over the entire data set, because it would have to know what axes are being reduced (and how).

This convention that "batches from the dataset need to be stackable along axis 0" is not enforced in the code because it assumes that data from a standardized dataset has the same dimension, only the batch size can vary.

To fix this, you would have to split up the test into a test for a 3d data set and a 4d data set. Let me know if this does not make sense, or if you think this assumption is unrealistic and we need to fix this in other parts of the library.

Reply:
Ok I see. I'm a bit confused that KFAC-reduce (type-2) matches the `curvlinops` GGN in the test in #63 for different weight-sharing dims per batch. Does this not violate the assumptions?

Reply:
That might be a lucky coincidence and I will try to think through this while reviewing #63. 

I think I've convinced myself that all `x` from a dataset must have the same shape and therefore my concern above is valid.
If this was not the case, we could not draw random data from the data set and stack them into a mini-batch (because stacking requires equal shape up until the axis to be stacked, which is the batch axis).